pconnector/config.go

```diff
@@ -0,0 +1,34 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pconnector
+
+// LogLevel represents a log level.
+type LogLevel int32
+
+const (
+	LogLevelNo    LogLevel = 0
+	LogLevelTrace LogLevel = 1
+	LogLevelDebug LogLevel = 2
+	LogLevelInfo  LogLevel = 3
+	LogLevelWarn  LogLevel = 4
+	LogLevelError LogLevel = 5
+	LogLevelOff   LogLevel = 6
+)
+
+type PluginConfig struct {
+	Token       string
+	ConnectorID string
+	LogLevel    LogLevel
```

Comment:
We could leave the log level as a string and let Conduit and the SDK handle the format 🤷 I imagine it's simplest to just use [`zerolog.Level.String`](https://pkg.go.dev/github.com/rs/zerolog#Level.String) and [`zerolog.ParseLevel`](https://pkg.go.dev/github.com/rs/zerolog#ParseLevel).

Reply:
I'd maybe stay away from `zerolog.Level` here because we might use some other logger in the future (looking at [`log/slog`](https://pkg.go.dev/log/slog) 👀), but more importantly, these values are coming from env variables so they are essentially just strings, we can let the SDK and Conduit interpret them.